### PR TITLE
Fixes and additions.

### PR DESCRIPTION
**Pull request by JakeDaBoss**
Nerfs, tweaks, fixes, the usual.

### CODE REVIEW

File: code/modules/admin/admin_verbs.dm [hunk 1]
```diff
@@ -326,6 +327,7 @@ var/list/admin_verbs_mentor = list(
 //	/client/proc/dsay,
 	/datum/admins/proc/paralyze_mob,
 	/client/proc/cmd_admin_subtle_message
```
Comment (Lykayos) [inline code/modules/admin/admin_verbs.dm]: Missing your comma here. Don't want to trigger the wonders of syntax errors.

File: code/_helpers/logging.dm [hunk 2]
```diff
@@ -36,6 +36,10 @@
 		if(C.is_preference_enabled(/datum/client_preference/debug/show_debug_logs))
 			C << "DEBUG: [text]"
 
+/proc/log_devsay(text)
+	if (config.log_devsay)
+		diary << "\[[time_stamp()]]DEVSAY: [text][log_end]"
```
Comment (BlueNexus) [inline code/_helpers/logging.dm]: This should follow the same format as the other logs. 
```
game_log("DEVSAY", text)
```

File: code/modules/admin/verbs/sendfax.dm [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+				P.stamped = new
+			P.stamped += /obj/item/weapon/stamp
+			P.overlays += stampoverlay
+			P.stamps += "<HR><i>This paper has been stamped[who=="NanoTrasen" ? "": " and encrypted"] by the [ who=="NanoTrasen" ? "Central Command":"Sol Government"] Quantum Relay.</i>"
+
+			usr << "Message reply to transmitted successfully."
+			log_and_message_admins("[key_name(usr)] sent a fax to [fax], as [who].")
+			return
+	else
+		usr<< "\red The fax is broken or has no power!"
```
Comment (Lykayos) [inline code/modules/admin/verbs/sendfax.dm]: Needs updated to span classes.

File: html/changelogs/JakeDaBoss.yml [hunk 4]
```diff
@@ -33,4 +33,11 @@ delete-after: True
 # Also, all entries are changed into a single [] after a master changelog generation. Just remove the brackets when you add new entries.
 # Please surround your changes in  double quotes ("), as certain characters otherwise screws up compiling. The quotes will not show up in the changelog.
 changes: 
-  - bugfix: "Corrected the accessible_z_levels list, which should correct a good deal of weirdness."
+  - bugfix: "Applied (hopefully) something that fixes all the stairs."
+  - bugfix: "Fixed unpowered APC in Paramedic's storage."
+  - bugfix: "Fixed a wrong wire in Upper Med."
+  - tweak: "Nerfed taser bolt's HALLOSS agony from 80 to 50."
```
Comment (BlueNexus) [inline html/changelogs/JakeDaBoss.yml]: Players won't know what HALLOSS means. Changelogs should be phrased in a way that normal players can understand.

File: code/modules/admin/verbs/sendfax.dm (hunk 3 above)
Comment (BlueNexus) [inline code/modules/admin/verbs/sendfax.dm]: Bad usage of \red

File: code/modules/projectiles/projectile/energy.dm [hunk 5]
```diff
@@ -67,7 +67,7 @@
 /obj/item/projectile/energy/electrode/stunshot
 	nodamage = 0
 	damage = 10
-	agony = 80
+	agony = 50
```
Comment (ghost) [inline code/modules/projectiles/projectile/energy.dm]: More of an opinion but this makes stun revolvers useless. They have a far slower fire rate, lack the ability to pass through objects, take up more space, have slow projectiles, and very few shots to a clip. Add in the fact that with this edit they hardly do more pain than a normal taser and there's no reason to bother with them anymore.

File: code/modules/admin/verbs/adminsay.dm [hunk 6]
```diff
@@ -34,5 +42,5 @@
 		sender_name = "<span class='admin'>[sender_name]</span>"
 	for(var/client/C in admins)
 		C << "<span class='mod_channel'>" + create_text_tag("mod", "MOD:", C) + " <span class='name'>[sender_name]</span>([admin_jump_link(mob, C.holder)]): <span class='message'>[msg]</span></span>"
-
+		C << "<span class='dev_channel'>" + create_text_tag("dev", "DEV:", C) + " <span class='name'>[sender_name]</span>(<A HREF='?src=\ref[C.holder];adminplayerobservejump=\ref[mob]'>JMP</A>): <span class='message'>[msg]</span></span>"
```
Comment (BlueNexus) [inline code/modules/admin/verbs/adminsay.dm]: This should follow the same format as mod_say and admin_say.

**Review by ghost — CHANGES_REQUESTED** [review]

**Comment by BlueNexus** [comment]
You have two reviews: Resolve them both.

File: code/modules/admin/admin_verbs.dm (hunk 1 above)
Comment (JakeDaBoss) [inline code/modules/admin/admin_verbs.dm]: aaaashit.

File: code/modules/projectiles/projectile/energy.dm (hunk 5 above)
Comment (JakeDaBoss) [reply code/modules/projectiles/projectile/energy.dm]: This also goes for the stun e-gun.

**Comment by ghost** [comment]
I like how our reviews happened within ten seconds of each other.

File: code/modules/projectiles/projectile/energy.dm (hunk 5 above)
Comment (ghost) [reply code/modules/projectiles/projectile/energy.dm]: Also as a note this is the projectile for the stun shotgun as well.

**Comment by BlueNexus** [comment]
I need confirmation that this will compile and pass bay's unit tests. I'm not convinced at the moment.

File: code/modules/projectiles/projectile/energy.dm (hunk 5 above)
Comment (ghost) [reply code/modules/projectiles/projectile/energy.dm]: The stun egun uses the /obj/item/projectile/beam/stun
This is the projectile for the research acquired stun revolver and the stun shells for the shotguns

Comment (JakeDaBoss) [reply code/modules/projectiles/projectile/energy.dm]: I'll cut some middle ground, make it 65.

Comment (ghost) [reply code/modules/projectiles/projectile/energy.dm]: What I'm saying is this isn't even the taser you edited.

Comment (ghost) [reply code/modules/projectiles/projectile/energy.dm]: Unless someone messed up the taser.

Comment (ghost) [reply code/modules/projectiles/projectile/energy.dm]: No just checked the taser is /beam/stun aswell

File: code/modules/admin/verbs/adminsay.dm [hunk 7]
```diff
@@ -42,5 +42,5 @@
 		sender_name = "<span class='admin'>[sender_name]</span>"
 	for(var/client/C in admins)
 		C << "<span class='mod_channel'>" + create_text_tag("mod", "MOD:", C) + " <span class='name'>[sender_name]</span>([admin_jump_link(mob, C.holder)]): <span class='message'>[msg]</span></span>"
-		C << "<span class='dev_channel'>" + create_text_tag("dev", "DEV:", C) + " <span class='name'>[sender_name]</span>(<A HREF='?src=\ref[C.holder];adminplayerobservejump=\ref[mob]'>JMP</A>): <span class='message'>[msg]</span></span>"
+		C << "<span class='dev_channel'>" + create_text_tag("dev", "DEV:", C) + " <span class='name'>[sender_name]</span>([admin_jump_link(mob, C.holder)]): <span class='message'>[msg]</span></span>"
```
Comment (BlueNexus) [inline code/modules/admin/verbs/adminsay.dm]: That's not what I meant. This section shouldn't be here. Look at how mod and adminsay does it.

File: code/modules/projectiles/projectile/energy.dm [hunk 8]
```diff
@@ -67,7 +67,7 @@
 /obj/item/projectile/energy/electrode/stunshot
 	nodamage = 0
 	damage = 10
-	agony = 50
+	agony = 65
```
Comment (BlueNexus) [inline code/modules/projectiles/projectile/energy.dm]: This still doesn't apply to taser stun or egun stun mode.

**Review by BlueNexus — CHANGES_REQUESTED** [review]

File: code/modules/projectiles/projectile/energy.dm (hunk 8 above)
Comment (JakeDaBoss) [reply code/modules/projectiles/projectile/energy.dm]: It apparantly does. We tested it. This goes for e-gun stun mode. It got Kaed down in one shot on 80.

File: code/modules/admin/verbs/sendfax.dm [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+				P.stamped = new
+			P.stamped += /obj/item/weapon/stamp
+			P.overlays += stampoverlay
+			P.stamps += "<HR><i>This paper has been stamped[who=="NanoTrasen" ? "": " and encrypted"] by the [ who=="NanoTrasen" ? "Central Command":"Sol Government"] Quantum Relay.</i>"
+
+			usr << "Message reply to transmitted successfully."
+			log_and_message_admins("[key_name(usr)] sent a fax to [fax], as [who].")
+			return
+	else
+		usr<< "<span class='warning'> The fax is broken or has no power!"
```
Comment (Lykayos) [inline code/modules/admin/verbs/sendfax.dm]: Close the span class too. ;-;

File: code/modules/admin/verbs/adminsay.dm [hunk 10]
```diff
@@ -28,19 +28,17 @@
...
+	msg = sanitize(msg)
+	log_admin("DEV: [key_name(src)] : [msg]")
 
 	var/sender_name = key_name(usr, 1)
 	if(check_rights(R_ADMIN, 0))
 		sender_name = "<span class='admin'>[sender_name]</span>"
 	for(var/client/C in admins)
 		C << "<span class='mod_channel'>" + create_text_tag("mod", "MOD:", C) + " <span class='name'>[sender_name]</span>([admin_jump_link(mob, C.holder)]): <span class='message'>[msg]</span></span>"
-		C << "<span class='dev_channel'>" + create_text_tag("dev", "DEV:", C) + " <span class='name'>[sender_name]</span>([admin_jump_link(mob, C.holder)]): <span class='message'>[msg]</span></span>"
 	feedback_add_details("admin_verb","MS") //If you are copy-pasting this, ensure the 2nd parameter is unique to the new proc!
```
Comment (BlueNexus) [inline code/modules/admin/verbs/adminsay.dm]: UUUGH
Lines 44 to 39 shouldn't be here, is what I'm trying to say.

File: code/modules/projectiles/projectile/energy.dm (hunk 8 above)
Comment (ghost) [reply code/modules/projectiles/projectile/energy.dm]: /obj/item/weapon/gun/energy/gun
    name = "energy gun"

[. . . ]

```
projectile_type = /obj/item/projectile/beam/stun
origin_tech = list(TECH_COMBAT = 3, TECH_MAGNET = 2)
modifystate = "energystun"

firemodes = list(
    list(mode_name="stun", projectile_type=/obj/item/projectile/beam/stun, modifystate="energystun"),
    list(mode_name="lethal", projectile_type=/obj/item/projectile/beam, modifystate="energykill"),
    )
```

File: code/modules/projectiles/projectile/energy.dm (hunk 5 above)
Comment (JakeDaBoss) [reply code/modules/projectiles/projectile/energy.dm]: What projectile does that use then?

Comment (ghost) [reply code/modules/projectiles/projectile/energy.dm]: /obj/item/projectile/beam/stun
Which deals 40 pain and downs someone in an average of three shots not counting armor.
Unless you headshot them, in that case it's around 2

**Comment by BlueNexus** [comment]
Confirm this compiles?

**Comment by JakeDaBoss** [comment]
@BlueNexus 
https://4stor.com/uwIT4
https://4stor.com/7mGuO

**Comment by BlueNexus** [comment]
Have these features been tested in their current state, especially devsay? (Also, this PR still needs to be squashed)

**Comment by JakeDaBoss** [comment]
Not yet. I can check tomorrow. How do I squash again, or you got a ref page for me?

**Comment by BlueNexus** [comment]
```
git rebase -i HEAD~[number of commits to squash]
```

Then change the "pick" beside each commit UNDER the top one to "s"
press [esc] [:wq] [enter]
Press [esc] [:wq] [enter] again on the next screen
then

```
git push -f origin Oliver-test-branch:Oliver-test-branch
```

**Comment by JakeDaBoss** [comment]
@BlueNexus

**Comment by JakeDaBoss** [comment]
Can't be squashed. My updates are too erratic. Code and map.

**Review by BlueNexus — CHANGES_REQUESTED** [review]
Please ensure that mapmerge is used. A 2500 add/remove diff is unacceptable.

**Comment by JakeDaBoss** [comment]
And you want me to do that post-edit now? This contains some pretty wanted fixes. This has been up for 3 days.

-------- Oorspronkelijk bericht --------
Van: BlueNexus notifications@github.com
Datum: 13-10-2016  00:45  (GMT+01:00)
Aan: JakeDaBoss/Novis-Station Novis-Station@noreply.github.com
Cc: Jake Miles visiongamingjake@hotmail.com, Author author@noreply.github.com
Onderwerp: Re: [JakeDaBoss/Novis-Station] Fixes and additions. (#50)

BlueNexus requested changes on this pull request.

Please ensure that mapmerge is used. A 2500 add/remove diff is unacceptable.

## 

You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub:
https://github.com/JakeDaBoss/Novis-Station/pull/50#pullrequestreview-3977150

**Comment by JakeDaBoss** [comment]
@bluenexus

I suggest merging this and stop nitpicking every detail. I'll work mapmerge in the next update to make the diff lesser.

It's seriously aggravating with how slow this goes, as I get all the questions how the update is coming.

**Comment by BlueNexus** [comment]
Using mapmerge is standard practice, I expected you to _know_ to do it. And here's an idea - Rather than bundling everything into one PR, which means it takes a long time to get merged, put separate things in separate PRs. 
I will allow the non-mapmerged map changes this once, but not again. Regardless, this cannot be merged until all requested changes are resolved, excluding the mapmerge one.

File: code/modules/projectiles/projectile/energy.dm (hunk 8 above)
Comment (BlueNexus) [reply code/modules/projectiles/projectile/energy.dm]: @JakeDaBoss The energy gun and taser don't _use_ /obj/item/projectile/energy/electrode/stunshot, they use /obj/item/projectile/beam/stun and /obj/item/projectile/beam.

**Comment by JakeDaBoss** [comment]
It won't be squashed. I'll take a look at the rest.

**Comment by JakeDaBoss** [comment]
@BlueNexus just merge it. I'm not squashing it, considering my commits can't be squashed.

**Comment by BlueNexus** [comment]
@jakedaboss I can't merge it while there are outstanding requested changes.

**Comment by JakeDaBoss** [comment]
Which are @bluenexus

**Review by ghost — APPROVED** [review]

**Review by BlueNexus — APPROVED** [review]

**Pull request merged**